Podcast/SearchViewController.swift

```diff
@@ -0,0 +1,151 @@
+//  Podcast
+//
+//  Created by Natasha Armbrust on 10/14/17.
+//  Copyright © 2017 Cornell App Development. All rights reserved.
+//
+
+import UIKit
+import SnapKit
+
+class SearchViewController: ViewController, UISearchControllerDelegate, UITableViewDelegate, UITableViewDataSource, TabbedViewControllerSearchResultsControllerDelegate {
+
+    var pastSearches: Set<String>? //keep around a set so we don't have duplicates
+    var searchController: UISearchController!
+    var searchResultsController: TabbedPageViewController!
+    var pastSearchesTableView: UITableView!
+    var tabUnderlineView: UnderlineTabBarView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
+        view.backgroundColor = .offWhite
+        
+        searchResultsController = TabbedPageViewController()
+        view.addSubview(searchResultsController.view)
+        
+        searchResultsController.searchResultsDelegate = self
+        
+        searchController = UISearchController(searchResultsController: searchResultsController)
+        searchController.searchResultsUpdater = searchResultsController
+
+        let cancelButtonAttributes: NSDictionary = [NSAttributedStringKey.foregroundColor: UIColor.sea]
+        UIBarButtonItem.appearance().setTitleTextAttributes(cancelButtonAttributes as? [NSAttributedStringKey: Any], for: .normal)
+        
+        searchController.searchBar.showsCancelButton = false
+        searchController.searchBar.searchBarStyle = .minimal
+        searchController.searchBar.placeholder = "Search"
+        searchController.delegate = self
+        searchController.hidesNavigationBarDuringPresentation = false
+        searchController.isActive = true
+        searchController.dimsBackgroundDuringPresentation = false
+        definesPresentationContext = true
+        
+        navigationItem.titleView = searchController?.searchBar
+        
+        pastSearchesTableView = UITableView(frame: CGRect.zero)
+        pastSearchesTableView.backgroundColor = .offWhite
+        pastSearchesTableView.backgroundView = EmptyStateView(type: .pastSearch)
+        pastSearchesTableView.backgroundView!.isHidden = true
+        pastSearchesTableView.register(PastSearchTableViewCell.self, forCellReuseIdentifier: "PastSearchCell")
+        pastSearchesTableView.delegate = self
+        pastSearchesTableView.dataSource = self
+        pastSearchesTableView.separatorStyle = .none
+        view.addSubview(pastSearchesTableView)
+        
+        pastSearchesTableView.snp.makeConstraints { make in
+            make.width.equalToSuperview()
+            make.top.equalToSuperview()
+            make.bottom.equalToSuperview().inset(appDelegate.tabBarController.tabBarHeight)
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        if let priorSearches = UserDefaults.standard.value(forKey: "PastSearches") as? [String] {
```

Comment:
UserDefaults is kinda slow this is something we probably want to use a better caching system for.

Reply:
Also this is definitely a privacy concern, past searches is sensitive information. Gotta think about the people looking up porn podcasts.

Reply:
^^ Drew!! What did you have in mind instead of UserDefaults? And for sure we should have a setting or tab to clear search history!

Reply:
Let's add this later on.

Reply:
Alright, leave it for now but definitely make a note that we should change it or we'll forget

Reply:
I will TODO.